File: backend/app/api/stats.py
from datetime import date

from fastapi import APIRouter

from app.api.deps import SessionDep
from app.schemas.stats import StatsResponse
from app.services.stats_service import compute_stats

router = APIRouter(prefix="/api/stats", tags=["stats"])


@router.get("", response_model=StatsResponse)
def get_stats(
    db: SessionDep,
    date_from: date | None = None,
    date_to: date | None = None,
    instrument_serial: str | None = None,
) -> StatsResponse:
    """Aggregated figures for the Stats dashboard. Time-series/throughput respect the
    date range (by run_date) and instrument filter; cell/sample/credit snapshots reflect
    current outstanding state - see services/stats_service.py for the scoping rules."""
    return compute_stats(db, date_from, date_to, instrument_serial)

File: frontend/src/api/stats.ts
import { api, buildQuery } from "./client";
import type { StatsResponse } from "@/types/stats";

export interface StatsParams {
  date_from?: string;
  date_to?: string;
  instrument_serial?: string;
}

export const statsApi = {
  get: (params: StatsParams = {}) => api.get<StatsResponse>(`/api/stats${buildQuery(params)}`),
};
